### PR TITLE
ROC-3020 Rename repository classes to avoid confusion

### PR DESCRIPTION
**Pull request by dharmendrak**
This PR is to achieve the following class structure.

nomnoml-2

This PR made following changes,
1) All search methods extracted from claim repository to legacySearchRepository are now merged back in claim repository.
2) Renamed existing CaseRepository implementer classes to DBCaseReposiory , CCDCaseRepository

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/search/CCDCaseDBI.java [hunk 1]
```diff
@@ -14,28 +14,28 @@
 
 import static java.lang.String.format;
 
-@Service("caseRepository")
+@Service("caseDBI")
 @ConditionalOnProperty(prefix = "core_case_data", name = "api.url")
-public class CCDCaseRepository implements CaseRepository {
-    private final Logger logger = LoggerFactory.getLogger(CCDCaseRepository.class);
+public class CCDCaseDBI implements CaseDBI {
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/search/CCDCaseDBI.java]: Repository is the spring boot name for the data access class

I would prefer:
keep ClaimRepository and revert this back to what it currently was

Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/search/CCDCaseDBI.java]: we will be discarding all claim names once we move to CCD, Claim->Case, ClaimService ->CaseService, and so fourth.

Comment (timja-kainos) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/search/CCDCaseDBI.java]: At that point ClaimRepository will be deleted

File: src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/ClaimStore.java [hunk 2]
```diff
@@ -25,10 +24,7 @@
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
     @Autowired
-    private ClaimRepository claimRepository;
-
-    @Autowired
-    private LegacyCaseRepository legacyCaseRepository;
+    private CaseDBI caseDBI;
```
Comment (timja-kainos) [inline src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/ClaimStore.java]: where has DBI come from?
Its not a spring term as far as I'm aware,
we use repository

**Comment by geek-soft** [comment]
Another possible variant: 

nomnoml-3

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java [hunk 3]
```diff
@@ -22,7 +22,7 @@
 
 @Service
 @ConditionalOnProperty(prefix = "core_case_data", name = "api.url")
-public class CCDClaimSearchRepository {
+public class CCDCaseApi {
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java]: The name is a bit unfortunate but I am happy o accept it for now as in the future the implementation is likely to be moved to `CCDCaseRepository`.

**Pull request closed without merging**